lib/protobuf/decoder.ex

```diff
@@ -14,6 +14,9 @@ defmodule Protobuf.Decoder do
     %{repeated_fields: repeated_fields} = msg_props = module.__message_props__()
     struct = build_struct(kvs, msg_props, module.new())
     reverse_repeated(struct, repeated_fields)
+  catch
+    :error, :function_clause ->
+      raise Protobuf.DecodeError, message: "cannot decode binary data as #{module}"
```

Comment:
If we just want to catch this kind of errors like your example, I prefer adding a catching-all pattern match for `raw_decode_value`.

Reply:
In this case the `function_clause` happens in `raw_decode_varint` though - should I instrument both functions?

Reply:
OK.